### PR TITLE
Skin Packs

### LINKED ISSUE #137: Major feat: Add-on support
**Is your feature request related to a problem? Please describe.**

Though resource modifications have always been "implicitly" supported [^1], it's never been a documented/encouraged feature. But it turns out that such modifications could customize each user's experience in a unique way.

**Describe the solution you'd like**

Add-on support. I mean, a way to support a secondary resource file that "superseeds" the main one. Of course, this will need a major refactor in our API to accept this "external source", but it's still possible.

A few other related ideas (that may come once we get the above things done):

- Support modifying the background color (which, by now, will always be black)
- Add the star appearance to the resource file, so these can also be customized.
- Distribute a sort of "official DLC bundle" for free to get y'all started! :smiley:

**Describe alternatives you've considered**

Just document the possibility of changing the main `resource.pyxres` file [^1]?

**Additional context**

Feel free to provide feedback or your thoughts on this feature!

And thanks to @HarshNarayanJha for providing the initial idea (via our Discord server), that is highly appreciated!

[^1]: You always had the possibility to modify `resource.pyxres`, which contains the main image sources, the original music, etc.

### PR DESCRIPTION
**Pull request by HarshNarayanJha**
Support for custom Skin Packs !!! 🎉 

Press 3 in the main menu and select a skin pack with live preview
Note that this commit contains a temporary resource file, for development purposes

closes #137

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -647,17 +651,32 @@ def draw_game(self):
     def update_menu(self):
         if pyxel.btnp(pyxel.KEY_Q):
             pyxel.quit()
-        if not self.menu_credits:
-            if pyxel.btnp(pyxel.KEY_1):
+        if not self.menu_credits and not self.menu_skin:
+            if pyxel.btnp(pyxel.KEY_1) or pyxel.btnp(pyxel.KEY_KP_1):
```
Comment (DiddiLeija) [inline main.py:655]: I'm just curious about this: what's the difference between `KEY_1` and `KEY_KP_1`?

File: main.py [hunk 2]
```diff
@@ -727,6 +749,28 @@ def draw_menu(self):
...
+            pyxel.text(
+                25, 35, "[1] Skin A" + ("<-" if self.current_skin == 0 else ""), 7
+            )
+            # Skin B
+            pyxel.text(
+                26, 45, "[2] Skin B" + ("<-" if self.current_skin == 1 else ""), 1
+            )
+            pyxel.text(
+                25, 45, "[2] Skin B" + ("<-" if self.current_skin == 1 else ""), 7
+            )
```
Comment (DiddiLeija) [inline main.py:770]: Just a small style fix:

```suggestion
            pyxel.text(
                26, 35, "[1] Skin A" + (" <-" if self.current_skin == 0 else ""), 1
            )
            pyxel.text(
                25, 35, "[1] Skin A" + (" <-" if self.current_skin == 0 else ""), 7
            )
            # Skin B
            pyxel.text(
                26, 45, "[2] Skin B" + (" <-" if self.current_skin == 1 else ""), 1
            )
            pyxel.text(
                25, 45, "[2] Skin B" + (" <-" if self.current_skin == 1 else ""), 7
            )

```

**Review by DiddiLeija — COMMENTED** [review]
Great work @HarshNarayanJha! :tada:
I've left a few thoughts below, but overall it's a great work.

File: main.py (hunk 1 above)
Comment (HarshNarayanJha) [reply main.py:655]: You don't know? Well the thing is my number keys are partially damaged. So I use the numpad one's (numeric keypad), and for that it is KEY_KP_1

Comment (DiddiLeija) [reply main.py:655]: Oh, that's right, thank you for adding these keys!

File: main.py (hunk 2 above)
Comment (HarshNarayanJha) [reply main.py:770]: Ok small spacing is good

File: main.py [hunk 3]
```diff
@@ -282,7 +282,10 @@ def __init__(self):
         # the app quits.
         self.messages = []
 
-        pyxel.load("resource.pyxres")
+        self.skins = ["resource.pyxres", "resource_2.pyxres"]
+        self.current_skin = 0
```
[the remark this answers is not in the document]
Comment (HarshNarayanJha) [reply main.py:286]: No for that, we will use os.listdir or grep.grep to get all pyxres files

**Comment by HarshNarayanJha** [comment]
@DiddiLeija I saw the comments. Will work on them tomorrow 😉

File: resource_diddi.pyxres
Comment (DiddiLeija) [inline resource_diddi.pyxres:1]: I may modify this name later (to consolidate it), but by now it's fine :)

**Review by DiddiLeija — APPROVED** [review]
Great work!
I think this PR is fully ready to be merged (I would like to work on documentation/style additions later) :)

File: resource_diddi.pyxres
Comment (HarshNarayanJha) [reply resource_diddi.pyxres:1]: Yea sure. I just thought that it wouldn't look no to be the default option be named "Default" 😂
That's why I renamed it to diddi, but you are obviously free to change it.

**Pull request merged**